File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1:
            return "1"
        previous = self.countAndSay(n-1)
        result = ""
        count = 1
        digit = previous[0]
        for i in range(1, len(previous)):
            if previous[i] == digit:
                count += 1
            else:
                result += str(count) + digit
                count = 1
                digit = previous[i]
        result += str(count) + digit
        return result
        

File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        
        string previous = countAndSay(n - 1);
        string result = "";
        int count = 1;
        char digit = previous[0];
        
        for (int i = 1; i < previous.size(); i++) {
            if (previous[i] == digit) {
                count++;
            } else {
                result += to_string(count) + digit;
                count = 1;
                digit = previous[i];
            }
        }
        
        result += to_string(count) + digit;
        return result;
    }
};
